civiclens/collect/access_api_data.py

```diff
@@ -167,20 +193,67 @@ def poll_for_response(api_key, wait_for_rate_reset):
+    if data_type == "comments":
+        all_comments = []
+        unique_comments = {}
+        
+        params.update({
+            "filter[commentOnId]": comment_on_id,
+            "page[size]": 250,
+            "sort": "lastModifiedDate,documentId"
+        })
+
+        last_modified_date = None
+        continue_fetching = True
+
+        while continue_fetching:
+            success, r_json = poll_for_response(api_key, wait_for_rate_reset=True)
+            if success:
+                all_comments.extend(r_json['data'])
+                print(f"Fetched {len(r_json['data'])} comments, total: {len(all_comments)}")
 
-        success, r_json = poll_for_response(api_key, wait_for_rate_reset=True)
+                # Check and handle the pagination
+                has_next_page = r_json["meta"].get("hasNextPage", False)
+                print(f"Has next page: {has_next_page}")
 
-        if success or (_is_duplicated_on_server(r_json) and skip_duplicates):
-            if doc_data is not None:
-                doc_data += r_json["data"]
+                if len(r_json['data']) < 250 or not has_next_page:
+                    print("No more pages or fewer than 250 comments fetched, stopping...")
+                    continue_fetching = False
+                else:
+                    last_modified_date = format_datetime_for_api(r_json['data'][-1]['attributes']['lastModifiedDate'])
+                    params = {
+                        "filter[commentOnId]": comment_on_id,
+                        "filter[lastModifiedDate][ge]": last_modified_date,
+                        "page[size]": 250,
+                        "sort": "lastModifiedDate,documentId",
+                        "page[number]": 1,
+                        "api_key": api_key
+                    }
+                    print(f"Fetching more data from {last_modified_date}")
             else:
-                doc_data = r_json["data"]
+                print("Failed to fetch data or API limit reached.")
+                continue_fetching = False
+
+        # Remove Duplicates
+        for comment in all_comments:
+            unique_comments[comment['id']] = comment
+        return list(unique_comments.values())
+    
+    else:
+        doc_data = None  # Initialize doc_data to None
+        for i in range(1, 21):  # Fetch up to 20 pages
+            params["page[number]"] = str(i)  # Add page number to the params
+
+            success, r_json = poll_for_response(api_key, wait_for_rate_reset=True)
+
+            if success or (_is_duplicated_on_server(r_json) and skip_duplicates):
+                if doc_data is not None:
+                    doc_data += r_json["data"]
+                else:
+                    doc_data = r_json["data"]
 
-            # Break if it's the last page
-            if r_json["meta"]["lastPage"]:
-                return doc_data
+                # Break if it's the last page
```

Comment:
I don't think this code should change indentation (?)

Reply:
Hi @andrewjtdunn, when you have a moment, could we schedule a call to discuss this? I'm not sure what you mean, and I'd appreciate your clarification.